Java/WhatIsNumberGame.java

```diff
@@ -0,0 +1,94 @@
+package whatnumber;
```

Comment:
Just update the readme of the Java folder with your entry, everything else LGTM :-)